torch/testing/_internal/common_modules.py

```diff
@@ -538,4 +602,14 @@ def module_inputs_torch_nn_MultiheadAttention(module_info, device, dtype, requir
                module_inputs_func=module_inputs_torch_nn_Embedding),
     ModuleInfo(torch.nn.ReLU,
                module_inputs_func=module_inputs_torch_nn_ReLU),
+    ModuleInfo(torch.nn.RNNCell,
+               module_inputs_func=partial(module_inputs_torch_nn_RNN_GRU_Cell, is_rnn=True)),
+    ModuleInfo(torch.nn.GRUCell,
+               module_inputs_func=module_inputs_torch_nn_RNN_GRU_Cell),
+    ModuleInfo(torch.nn.LSTMCell,
+               module_inputs_func=module_inputs_torch_nn_LSTMCell,
+               decorators=(
+                   DecorateInfo(unittest.expectedFailure, 'TestModule', 'test_grad'),
+                   DecorateInfo(unittest.expectedFailure, 'TestModule', 'test_gradgrad'),
```

Comment:
Any idea why these fail? Can you please add some comments here for context?

Reply:
I am not really sure. Will open an issue and add the link in the comment.

Reply:
https://github.com/pytorch/pytorch/issues/70362#issuecomment-1000663683

gradcheck and gradgradcheck don't like nested sequence of inputs.

With that in mind, I updated test modules with the following patch which works. Let me know if we are ok using pytree for this.

```patch
diff --git a/test/test_modules.py b/test/test_modules.py
index 7e1fe7aa88..9d3cbd3ac0 100644
--- a/test/test_modules.py
+++ b/test/test_modules.py
@@ -430,7 +430,10 @@ class TestModule(TestCase):
 
             grad_input = input_args + params + tuple(obj for (_, obj) in kwarg_tensors)
 
-            def fn_to_gradcheck(*input_and_params):
+            flat_input, flat_spec = torch.utils._pytree.tree_flatten(grad_input)
+
+            def fn_to_gradcheck(*flat_input_and_params):
+                input_and_params = torch.utils._pytree.tree_unflatten(flat_input_and_params, flat_spec)
                 new_input_args = input_and_params[:len(input_args)]
                 kwarg_args = input_and_params[-len(kwarg_tensors):]
                 new_kwargs = {name: obj for (name, _), obj in zip(kwarg_tensors, kwarg_args)}
@@ -438,7 +441,7 @@ class TestModule(TestCase):
                 with freeze_rng_state():
                     return m(*new_input_args, **new_kwargs, **other_kwargs)
 
-            self.assertTrue(check(fn_to_gradcheck, grad_input))
+            self.assertTrue(check(fn_to_gradcheck, flat_input))
 
```

Reply:
Have applied the above patch

Reply:
Thanks for digging into this! Flattening seems like a good solution to me